webui/ocw/lib/EC2db.py

```diff
@@ -35,17 +35,18 @@ def _instance_to_json(i):
 
 
 @transaction.atomic
-def sync_instances_db(region, instances):
+def sync_instances_db(region, instances, vault_namespace):
```

Comment:
why we not sync all namespaces in one shot ?

Reply:
You need special credentials for each namespace.

Reply:
how is that related ? if you not modify `sync_instances_db` method to take into account namespace this would lead to not filtering by namespace in line 40 which would lead that loop in line 43 will send to update_or_create_instance **all** available instances of EC2 ( each one still having its own credentials )

Reply:
The instance doesn't know to which "vault-namespace" it belongs to, thats why I'm adding it here.
We could also hook the get_instances() methods to add this parameter directly into the instance. Or let get_instances() directly return `ocw.models.Instance` objects.

Reply:
> The instance doesn't know to which "vault-namespace" it belongs to, thats why I'm adding it here.

if instance don't know "vault-namespace" how this is possible ? : 
`o = o.filter(region=region, provider=ProviderChoice.EC2, vault_namespace=vault_namespace)`

Reply:
you need to differ between two types of instance. We have a local one in our DB this is what you see here and the one in the CSP which doesn't know vault-namespaces.

Reply:
1. does not matter , what matters is that this `o` which is result of filter method suppose to know about namespaces, so this give me confidence to say that method sync_instances_db may exists without input parameter `namespaces ` 
2.  didn't get how the one that is in CSP is related to this function ? can you point me to line in repository where we trying to query instance  object **in** CSP ?

Reply:
1) The problem is that you get a subset of all instances in db from the provider. And this subset has common parameters which differs to other subsets. And one of these keys are vault_namespace.

2.)Sure, it is always a similar pattern.
 **Azure:** 
foreach vault_namespaces do:
   i = Azure(namespace)->[get_instances_from_csp](https://github.com/cfconrad/pcw/pull/34/files#diff-0ad2fc8593cc924b952fd5277e6a11d7R79);
   [sync_instances_with_local_db(i,ns)](https://github.com/cfconrad/pcw/pull/34/files#diff-0ad2fc8593cc924b952fd5277e6a11d7R81)

**Ec2** (It's a bit different, as the instance didn't contain the region, but we need the region to destroy the instance). Thats why it is:
foreach vault namespace
   foreach region
       i = EC2(region, namespace)->[get_instances_from_csp](https://github.com/cfconrad/pcw/pull/34/files#diff-0ad2fc8593cc924b952fd5277e6a11d7R85)
       [sync_instances_with_local_db(i,ns)](https://github.com/cfconrad/pcw/pull/34/files#diff-0ad2fc8593cc924b952fd5277e6a11d7R87)